lib/zeus/api_client/rest_interface.rb

```diff
@@ -21,20 +21,49 @@
 module RestInterface
   private
 
-  def get(path, params = {})
-    RestClient.get "#{@endpoint}#{path}", params: params
+  def get(path, token, bucket_name, params = {})
```

Comment:
not sure its this class responsibility to create the auth header from the bucket_name and token, maybe create a struct that is built from these parameters that creates this header, and then pass it to this class as params?

Theres repeated code in every call for building this header

What do you think?